src/Loader/FileLoader.php

```diff
@@ -40,25 +42,33 @@ public function loadClass(MapperMetadata $mapperMetadata): void
         $className = $mapperMetadata->className;
         $classPath = $this->directory . \DIRECTORY_SEPARATOR . $className . '.php';
 
-        if ($this->reloadStrategy === FileReloadStrategy::NEVER && file_exists($classPath)) {
-            require $classPath;
+        // We lock the file here, because another process could be writing the file at the same time
```

Comment:
Shouldn't we wait until the _other_ process is done until returning this method ?
So we are sure the file is generated.